mlflow/pytorch/torchserve/deploy/serve.py

```diff
@@ -0,0 +1,264 @@
+        if resp.status_code != 200:
+            raise Exception("Unable to infer the results")
+
+        if output is not None and len(output) != 0:
+            output_path = str(output[0])
+            if os.path.exists(output_path):
+                with open(output_path + "/output.json", "w") as fp:
+                    json.dump({"result": resp.text}, fp)
+            else:
+                with open("output.json", "w") as fp:
+                    json.dump({"result": resp.text}, fp)
+        return resp.text
+
+    def generate_mar_file(
+        self, model_name, version, model_file, handler_file, extra_files, model_uri
+    ):
+
+        """
+        Generates mar file using the torch archiver in the specified model store path
+        """
+
+        if not os.path.isfile(model_uri):
+            model_path = Path(_download_artifact_from_uri(model_uri))
+            model_uri = model_path
+
+        export_path = self.server_config["export_path"]
+        if export_path:
+            model_store = export_path
+        else:
+            model_store = "model_store"
+            if not os.path.isdir(model_store):
+                os.makedirs(model_store)
+
+        cmd = (
+            "torch-model-archiver --force --model-name {} "
+            "--version {} --model-file {} --serialized-file {} "
+            "--handler {} --export-path {}".format(
+                model_name, version, model_file, model_uri, handler_file, model_store
+            )
+        )
+        if extra_files:
+            cmd = "{} --extra-files {}".format(cmd, extra_files)
+
+        return_code = os.system(cmd)
+        if return_code != 0:
+            _logger.error(
+                "Error when attempting to load and parse JSON cluster spec from file %s",
+                cmd,
+            )
+            raise Exception("Unable to create mar file")
+
+        if export_path:
+            mar_file = "{}/{}.mar".format(export_path, model_name)
+        else:
+            mar_file = "{}/{}/{}.mar".format(os.getcwd(), model_store, model_name)
+            if os.path.isfile(mar_file):
+                print("{} file generated successfully".format(mar_file))
+        return mar_file
+
+    def register_model(
+        self, mar_file_path, name, model_uri=None, flavor=None, config=None
+    ):
+        """
+        Register the model using the mar file that has been generated by the archiver
+        """
+        query_path = mar_file_path
+        if config is not None:
+            for key in config:
+                query_path += "&" + key + "=" + str(config[key])
+        else:
+            query_path += "&initial_workers=" + str(1)
+
+        url = "{}/{}?url={}".format(self.management_api, "models", query_path)
+        resp = requests.post(url=url)
+
+        if resp.status_code != 200:
+            raise Exception("Unable to register the model")
+        return True
+
+
+def run_local(name, model_uri, flavor=None, config=None):
+    pass
```

Comment:
It might better to throw here, as opposed to performing a no-op

Reply:
Yes. we will implement this method . We will send a PR for this soon